### PR TITLE
Refactor global variables to autoload variables

### PR DESCRIPTION
**Pull request by lucc**
This is just an idea. If you like it I will continue to work on it. See commit message for more.

### CODE REVIEW

**Comment by lucc** [comment]
I have rebased (actually rewritten) these commits on top of the current master. I have also split the work into different commits to better show what is done.

**Comment by gmarik** [comment]
@lucc looks much nicer!

File: autoload/vundle.vim [hunk 1]
```diff
@@ -61,24 +61,30 @@ endif
...
 
 " Finishes putting plugins on the rtp.
 func! vundle#end(...) abort
-  unlet g:vundle_lazy_load
+  unlet g:vundle#lazy_load
   call vundle#config#activate_bundles()
 endf
 
+" Initialize some global variables used by Vundle.
+let vundle#bundle_dir = expand('$HOME/.vim/bundle', 1)
```
Comment (jdevera) [inline autoload/vundle.vim]: Curious here: Why bother giving this a sensible value when we **always** will initialise it in `#rc`?

**Comment by lucc** [comment]
@jdevera I tried to apply your comment (sadly it is lost now, as the commit is gone)

**Comment by jdevera** [comment]
I wish I remembered that comment :)

**Comment by jdevera** [comment]
I gave this a bit of a run and it seems solid to me. Merging.

**Pull request merged**